tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py

```diff
@@ -625,6 +630,59 @@ def _InputName(inp):
         if k not in removed_const_nodes
     }
 
+    def dependency_optimization(input_map):
```

Comment:
How would you compare adding this with the following two approaches:
1. do not add those grappler passed to TF1.
2. change the test to expect different result for TF1.

Reply:
1. Do you mean in the test, or in the PR in general? This is not exactly a TF1 problem, it just messes up the TF1 tests because they do a thorough check of edges in the generated graph. Apparently, when the v2 tests were added, these checks were not working because of control dependencies working differently, so the graph checks are less strict with v2 (it checks that segments contain the expected nodes). Actually, I find these tests (both v1 and v2) too strict, because they're very sensitive to how grappler optimizes the graph, and an equivalent but different graph can make tests fail (that's all these failures we've been seeing).
2. I first considered that instead of my fix. If we make sure to avoid all redundant control dependencies in `ControlDependencyTest` (essentially, just the first node of a segment has a control dependency), and use data dependencies in other tests, it should pass. But it means we leave an incorrect `_VerifyConnections` check, likely to break with any test that uses control dependencies in the future.

II understand if you think that my fix is a bit too much though, I can revert and change the tests to avoid this problem.

Reply:
@bixia1 Can you please assist on above comments from @Nyrio. Thanks!

Reply:
@bixia1 What about this: I can change the tests so that for a handful of tests, including `ControlDependencyTest`, the expected edges are provided explicitly. For other tests, `VerifyConnections` is disabled (like in the V2 tests).
Additionally, I could try enabling the modified `VerifyConnections` check in some V2 tests too.

Reply:
Sound good to me.